pytorch_lightning/trainer/trainer.py

```diff
@@ -942,8 +943,14 @@ def _run(self, model: "pl.LightningModule") -> Optional[Union[_EVALUATE_OUTPUT,
         if self.state.fn == TrainerFn.FITTING:
             self.call_hook("on_fit_end")
 
-        # teardown
-        self._call_teardown_hook(model)
+        # teardown if necessary (similar calls for spawn plugins are excluded as they have
+        # been included at the end of `new_process` functions)
+        if self._distrib_type not in (
+            DistributedType.DDP_SPAWN,
+            DistributedType.TPU_SPAWN,
+            DistributedType.DDP_SHARDED_SPAWN,
+        ):
```

Comment:
```suggestion
        # teardown if necessary (similar calls for spawn plugins are excluded as they have
        # been included at the end of `new_process` functions)
        if not self._distrib_type.is_interactive_compatible():
```
https://github.com/PyTorchLightning/pytorch-lightning/blob/9ca02f58aeba6427f1a45b18818307e95b3f4ac1/pytorch_lightning/utilities/enums.py#L73-L76